Lib/importlib/_bootstrap.py

```diff
@@ -85,6 +85,8 @@ def __enter__(self):
     def __exit__(self, *args, **kwargs):
         """Remove self.lock from this thread's _blocking_on list."""
         self.blocked_on.remove(self.lock)
+        if not self.blocked_on:
+            del _blocking_on[self.thread_id]
```

Comment:
Just have one question, is this always true?
```py
assert self.blocked_on is _blocking_on[self.thread_id]
```

Reply:
I think, if it were sometimes is not true, `self.blocked_on.remove(...)` will fail, no?
Though, currently test suite didn't failed with new assert check.

Reply:
See https://github.com/python/cpython/pull/101942/files/79fd697175682f153504534aa2320a1ebf3fd8fa#r1109686308

It may not be True under certain circumstances.